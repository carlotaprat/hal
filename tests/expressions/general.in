a = 3+2 + 1-2
b = -(a)
b = -a

f = -(f 2)
f = 3-2+f
f = a-b
f = a -b
f = a - b

f = a - b - c 3, 4 - d -e, -f