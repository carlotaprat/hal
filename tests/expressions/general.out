digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FUNCALL"];
  n1 [label="FUNCALL"];
  n2 [label="a"];
  n3 [label="ARGS"];
  n4 [label="="];
  n5 [label="-"];
  n5 [label="-"];
  n6 [label="+"];
  n6 [label="+"];
  n7 [label="+"];
  n7 [label="+"];
  n8 [label="3"];
  n9 [label="2"];
  n10 [label="1"];
  n11 [label="2"];
  n12 [label="FUNCALL"];
  n12 [label="FUNCALL"];
  n13 [label="b"];
  n14 [label="ARGS"];
  n15 [label="="];
  n16 [label="NEGATE"];
  n16 [label="NEGATE"];
  n17 [label="FUNCALL"];
  n17 [label="FUNCALL"];
  n18 [label="a"];
  n19 [label="ARGS"];
  n20 [label="FUNCALL"];
  n20 [label="FUNCALL"];
  n21 [label="b"];
  n22 [label="ARGS"];
  n23 [label="="];
  n24 [label="NEGATE"];
  n24 [label="NEGATE"];
  n25 [label="FUNCALL"];
  n25 [label="FUNCALL"];
  n26 [label="a"];
  n27 [label="ARGS"];
  n28 [label="FUNCALL"];
  n28 [label="FUNCALL"];
  n29 [label="f"];
  n30 [label="ARGS"];
  n31 [label="="];
  n32 [label="NEGATE"];
  n32 [label="NEGATE"];
  n33 [label="FUNCALL"];
  n33 [label="FUNCALL"];
  n34 [label="f"];
  n35 [label="ARGS"];
  n35 [label="ARGS"];
  n36 [label="2"];
  n37 [label="FUNCALL"];
  n37 [label="FUNCALL"];
  n38 [label="f"];
  n39 [label="ARGS"];
  n40 [label="="];
  n41 [label="+"];
  n41 [label="+"];
  n42 [label="-"];
  n42 [label="-"];
  n43 [label="3"];
  n44 [label="2"];
  n45 [label="FUNCALL"];
  n45 [label="FUNCALL"];
  n46 [label="f"];
  n47 [label="ARGS"];
  n48 [label="FUNCALL"];
  n48 [label="FUNCALL"];
  n49 [label="f"];
  n50 [label="ARGS"];
  n51 [label="="];
  n52 [label="-"];
  n52 [label="-"];
  n53 [label="FUNCALL"];
  n53 [label="FUNCALL"];
  n54 [label="a"];
  n55 [label="ARGS"];
  n56 [label="FUNCALL"];
  n56 [label="FUNCALL"];
  n57 [label="b"];
  n58 [label="ARGS"];
  n59 [label="FUNCALL"];
  n59 [label="FUNCALL"];
  n60 [label="f"];
  n61 [label="ARGS"];
  n62 [label="="];
  n63 [label="FUNCALL"];
  n63 [label="FUNCALL"];
  n64 [label="a"];
  n65 [label="ARGS"];
  n65 [label="ARGS"];
  n66 [label="NEGATE"];
  n66 [label="NEGATE"];
  n67 [label="FUNCALL"];
  n67 [label="FUNCALL"];
  n68 [label="b"];
  n69 [label="ARGS"];
  n70 [label="FUNCALL"];
  n70 [label="FUNCALL"];
  n71 [label="f"];
  n72 [label="ARGS"];
  n73 [label="="];
  n74 [label="-"];
  n74 [label="-"];
  n75 [label="FUNCALL"];
  n75 [label="FUNCALL"];
  n76 [label="a"];
  n77 [label="ARGS"];
  n78 [label="FUNCALL"];
  n78 [label="FUNCALL"];
  n79 [label="b"];
  n80 [label="ARGS"];
  n81 [label="FUNCALL"];
  n81 [label="FUNCALL"];
  n82 [label="f"];
  n83 [label="ARGS"];
  n84 [label="="];
  n85 [label="-"];
  n85 [label="-"];
  n86 [label="-"];
  n86 [label="-"];
  n87 [label="FUNCALL"];
  n87 [label="FUNCALL"];
  n88 [label="a"];
  n89 [label="ARGS"];
  n90 [label="FUNCALL"];
  n90 [label="FUNCALL"];
  n91 [label="b"];
  n92 [label="ARGS"];
  n93 [label="FUNCALL"];
  n93 [label="FUNCALL"];
  n94 [label="c"];
  n95 [label="ARGS"];
  n95 [label="ARGS"];
  n96 [label="3"];
  n97 [label="-"];
  n97 [label="-"];
  n98 [label="4"];
  n99 [label="FUNCALL"];
  n99 [label="FUNCALL"];
  n100 [label="d"];
  n101 [label="ARGS"];
  n101 [label="ARGS"];
  n102 [label="NEGATE"];
  n102 [label="NEGATE"];
  n103 [label="FUNCALL"];
  n103 [label="FUNCALL"];
  n104 [label="e"];
  n105 [label="ARGS"];
  n106 [label="NEGATE"];
  n106 [label="NEGATE"];
  n107 [label="FUNCALL"];
  n107 [label="FUNCALL"];
  n108 [label="f"];
  n109 [label="ARGS"];

  n0 -> n1 // "BLOCK" -> "FUNCALL"
  n1 -> n2 // "FUNCALL" -> "a"
  n1 -> n3 // "FUNCALL" -> "ARGS"
  n0 -> n4 // "BLOCK" -> "="
  n0 -> n5 // "BLOCK" -> "-"
  n5 -> n6 // "-" -> "+"
  n6 -> n7 // "+" -> "+"
  n7 -> n8 // "+" -> "3"
  n7 -> n9 // "+" -> "2"
  n6 -> n10 // "+" -> "1"
  n5 -> n11 // "-" -> "2"
  n0 -> n12 // "BLOCK" -> "FUNCALL"
  n12 -> n13 // "FUNCALL" -> "b"
  n12 -> n14 // "FUNCALL" -> "ARGS"
  n0 -> n15 // "BLOCK" -> "="
  n0 -> n16 // "BLOCK" -> "NEGATE"
  n16 -> n17 // "NEGATE" -> "FUNCALL"
  n17 -> n18 // "FUNCALL" -> "a"
  n17 -> n19 // "FUNCALL" -> "ARGS"
  n0 -> n20 // "BLOCK" -> "FUNCALL"
  n20 -> n21 // "FUNCALL" -> "b"
  n20 -> n22 // "FUNCALL" -> "ARGS"
  n0 -> n23 // "BLOCK" -> "="
  n0 -> n24 // "BLOCK" -> "NEGATE"
  n24 -> n25 // "NEGATE" -> "FUNCALL"
  n25 -> n26 // "FUNCALL" -> "a"
  n25 -> n27 // "FUNCALL" -> "ARGS"
  n0 -> n28 // "BLOCK" -> "FUNCALL"
  n28 -> n29 // "FUNCALL" -> "f"
  n28 -> n30 // "FUNCALL" -> "ARGS"
  n0 -> n31 // "BLOCK" -> "="
  n0 -> n32 // "BLOCK" -> "NEGATE"
  n32 -> n33 // "NEGATE" -> "FUNCALL"
  n33 -> n34 // "FUNCALL" -> "f"
  n33 -> n35 // "FUNCALL" -> "ARGS"
  n35 -> n36 // "ARGS" -> "2"
  n0 -> n37 // "BLOCK" -> "FUNCALL"
  n37 -> n38 // "FUNCALL" -> "f"
  n37 -> n39 // "FUNCALL" -> "ARGS"
  n0 -> n40 // "BLOCK" -> "="
  n0 -> n41 // "BLOCK" -> "+"
  n41 -> n42 // "+" -> "-"
  n42 -> n43 // "-" -> "3"
  n42 -> n44 // "-" -> "2"
  n41 -> n45 // "+" -> "FUNCALL"
  n45 -> n46 // "FUNCALL" -> "f"
  n45 -> n47 // "FUNCALL" -> "ARGS"
  n0 -> n48 // "BLOCK" -> "FUNCALL"
  n48 -> n49 // "FUNCALL" -> "f"
  n48 -> n50 // "FUNCALL" -> "ARGS"
  n0 -> n51 // "BLOCK" -> "="
  n0 -> n52 // "BLOCK" -> "-"
  n52 -> n53 // "-" -> "FUNCALL"
  n53 -> n54 // "FUNCALL" -> "a"
  n53 -> n55 // "FUNCALL" -> "ARGS"
  n52 -> n56 // "-" -> "FUNCALL"
  n56 -> n57 // "FUNCALL" -> "b"
  n56 -> n58 // "FUNCALL" -> "ARGS"
  n0 -> n59 // "BLOCK" -> "FUNCALL"
  n59 -> n60 // "FUNCALL" -> "f"
  n59 -> n61 // "FUNCALL" -> "ARGS"
  n0 -> n62 // "BLOCK" -> "="
  n0 -> n63 // "BLOCK" -> "FUNCALL"
  n63 -> n64 // "FUNCALL" -> "a"
  n63 -> n65 // "FUNCALL" -> "ARGS"
  n65 -> n66 // "ARGS" -> "NEGATE"
  n66 -> n67 // "NEGATE" -> "FUNCALL"
  n67 -> n68 // "FUNCALL" -> "b"
  n67 -> n69 // "FUNCALL" -> "ARGS"
  n0 -> n70 // "BLOCK" -> "FUNCALL"
  n70 -> n71 // "FUNCALL" -> "f"
  n70 -> n72 // "FUNCALL" -> "ARGS"
  n0 -> n73 // "BLOCK" -> "="
  n0 -> n74 // "BLOCK" -> "-"
  n74 -> n75 // "-" -> "FUNCALL"
  n75 -> n76 // "FUNCALL" -> "a"
  n75 -> n77 // "FUNCALL" -> "ARGS"
  n74 -> n78 // "-" -> "FUNCALL"
  n78 -> n79 // "FUNCALL" -> "b"
  n78 -> n80 // "FUNCALL" -> "ARGS"
  n0 -> n81 // "BLOCK" -> "FUNCALL"
  n81 -> n82 // "FUNCALL" -> "f"
  n81 -> n83 // "FUNCALL" -> "ARGS"
  n0 -> n84 // "BLOCK" -> "="
  n0 -> n85 // "BLOCK" -> "-"
  n85 -> n86 // "-" -> "-"
  n86 -> n87 // "-" -> "FUNCALL"
  n87 -> n88 // "FUNCALL" -> "a"
  n87 -> n89 // "FUNCALL" -> "ARGS"
  n86 -> n90 // "-" -> "FUNCALL"
  n90 -> n91 // "FUNCALL" -> "b"
  n90 -> n92 // "FUNCALL" -> "ARGS"
  n85 -> n93 // "-" -> "FUNCALL"
  n93 -> n94 // "FUNCALL" -> "c"
  n93 -> n95 // "FUNCALL" -> "ARGS"
  n95 -> n96 // "ARGS" -> "3"
  n95 -> n97 // "ARGS" -> "-"
  n97 -> n98 // "-" -> "4"
  n97 -> n99 // "-" -> "FUNCALL"
  n99 -> n100 // "FUNCALL" -> "d"
  n99 -> n101 // "FUNCALL" -> "ARGS"
  n101 -> n102 // "ARGS" -> "NEGATE"
  n102 -> n103 // "NEGATE" -> "FUNCALL"
  n103 -> n104 // "FUNCALL" -> "e"
  n103 -> n105 // "FUNCALL" -> "ARGS"
  n101 -> n106 // "ARGS" -> "NEGATE"
  n106 -> n107 // "NEGATE" -> "FUNCALL"
  n107 -> n108 // "FUNCALL" -> "f"
  n107 -> n109 // "FUNCALL" -> "ARGS"

}

